ReactiveUI.XamForms/XamForms/ActivationForViewFetcher.cs

```diff
@@ -32,17 +31,61 @@ public IObservable<bool> GetActivationForView(IActivatable view)
                     Observable.FromEventPattern<EventHandler, EventArgs>(x => page.Appearing += x, x => page.Appearing -= x).Select(_ => true),
                     Observable.FromEventPattern<EventHandler, EventArgs>(x => page.Disappearing += x, x => page.Disappearing -= x).Select(_ => false));
             } else {
-                var cell = view as Cell;
+                var xfView = view as View;
 
-                if (cell != null) {
-                    ret = Observable
-                        .Merge(
-                            Observable.FromEventPattern<EventHandler, EventArgs>(x => cell.Appearing += x, x => cell.Appearing -= x).Select(_ => true),
-                            Observable.FromEventPattern<EventHandler, EventArgs>(x => cell.Disappearing += x, x => cell.Disappearing -= x).Select(_ => false));
+                if (xfView != null) {
+                    var propertyChanged = Observable.FromEventPattern<PropertyChangedEventHandler, PropertyChangedEventArgs>(
+                        x => xfView.PropertyChanged += x,
+                        x => xfView.PropertyChanged -= x);
+                    var parentChanged = propertyChanged
+                        .Where(x => x.EventArgs.PropertyName == "Parent")
+                        .Select(_ => Unit.Default);
+
+                    return parentChanged
+                        .StartWith(Unit.Default)
+                        .Select(_ => GetPageFor(xfView))
+                        .Select(x =>
+                            x == null ?
+                                Observable.Return(false) :
+                                Observable
+                                    .Merge(
+                                        Observable.FromEventPattern<EventHandler, EventArgs>(y => x.Appearing += y, y => x.Appearing -= y).Select(_ => true),
```

Comment:
Not really sure we care about performance here, but if we do, there's a `FromEvent` overload which performs much better than `FromEventPattern` as it uses no reflection ([benchmark](https://gist.github.com/gluck/da82c1b537e44b55daa8)), and isn't really more complex to write:

instead of:

    Observable.FromEventPattern<EventHandler, EventArgs>(y => x.Appearing += y, y => x.Appearing -= y).Select(_ => true)

You can write:

    Observable.FromEvent<EventHandler, bool>(a => (_, __) => a(true), y => x.Appearing += y, y => x.Appearing -= y)